plutus-tx/src/Language/PlutusTx/AssocMap.hs

```diff
@@ -53,6 +54,12 @@ instance Functor (Map k) where
             go ((c, i):xs') = (c, f i) : go xs'
         in Map (go mp)
 
+instance (Read k, Read e) => Read (Map k e) where
```

Comment:
Fine, but add a note that this is only used by Marlowe and should be removed once you have proper serialization going.